clusterclaims/apply.sh

```diff
@@ -175,50 +182,86 @@ else
 fi
 
 
+#-----DETECT IF THE USER IS USING A SERVICE ACCOUNT - IF SO SET SUBJECT-----#
+if [[ $(oc whoami | awk -F ":" '{print $2}') == "serviceaccount" ]]; then
+    printf "${GREEN}* ServiceAccount use Detected, automatically adding ServiceAccount as a Subject${CLEAR}\n"
+    CLUSTERCLAIM_SERVICE_ACCOUNT="$(oc whoami | awk -F ":" '{print $4}')"
+    echo "" >> ./${CLUSTERCLAIM_NAME}/${CLUSTERCLAIM_NAME}.clusterclaim.yaml
+    ${SED} -e "s/__RBAC_SERVICEACCOUNT_NAME__/$CLUSTERCLAIM_SERVICE_ACCOUNT/g" \
+        -e "s/__CLUSTERCLAIM_NAMESPACE__/$CLUSTERPOOL_TARGET_NAMESPACE/g" \
+        ./templates/clusterclaim.subject.serviceaccount.yaml.template >> ./${CLUSTERCLAIM_NAME}/${CLUSTERCLAIM_NAME}.clusterclaim.yaml
+fi
+
+
 #-----OPTIONALLY SELECT AN RBAC GROUP-----#
 if [[ "$CLUSTERCLAIM_GROUP_NAME" == "" ]]; then
     printf  "${BLUE}- note: if you choose 'Y', you must have read permissions on group.user.openshift.io.${CLEAR}\n"
     printf "${YELLOW}Do you want to associate this ClusterClaim with an RBAC Group? (Y/N) ${CLEAR}"
     read selection
     if [[ "$selection" == "Y" || "$selection" == "y" ]]; then
-        # Prompt the user to choose a project
-        groups=$(oc get group -o=custom-columns=NAME:.metadata.name)
-        group_names=()
-        i=0
-        IFS=$'\n'
-        for line in $groups; do
-            if [ $i -eq 0 ]; then
-                printf "   \t$line\n"
-            else
-                printf "($i)\t$line\n"
-                unset IFS
-                line_list=($line)
-                group_names+=(${line_list[0]})
-                IFS=$'\n'
+        # Prompt the user to choose an RBAC GROUP
+        groups=$(oc get group -o=custom-columns=NAME:.metadata.name 2> /dev/null)
+        if [[ "$?" != "0" ]]; then
+            printf "${BLUE}- It looks like you don't have any RBAC groups (our query errored).${CLEAR}\n"
```

Comment:
Maybe instead of "have any" we say, "have access to any"?

Reply:
That's a good point, one sec!